src/app/roles.tsx

```diff
@@ -13,12 +13,18 @@ type RolesInterface = {
     [key in Roles]: string[];
 };
 
-const pathsShownToCaller = ["/calendar", "/clients", "/parcels", "/update-password"];
+const pathsShownToCaller = [
+    "/calendar",
+    "/clients",
+    "/parcels",
+    "/update-password",
+    "/set-password",
+];
 
 const roleToShownPages: RolesInterface = {
     admin: pathsShownToCaller.concat("/admin", "/lists"),
     caller: pathsShownToCaller,
-    "": ["/login", "/forgot-password", "/auth/reset-password"],
+    "": ["/login", "/forgot-password", "/auth/reset-password", "/set-password"],
```

Comment:
seems like this list serves the same purpose as `pathsNotRequiringLogin` in auth.tsx. Can we commonise this somehow? (keeping the list `as const` might be a challenge though)